server/neptune/gateway/event/local/repo_generator.go

```diff
@@ -0,0 +1,94 @@
+package local
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const workingDirPrefix = "repos"
+
+// GitRepoGenerator implements RepoGenerator through git commands
+type GitRepoGenerator struct {
```

Comment:
not to be that guy, but `GitRepoGenerator` sounds like the repo is being generated (ie. created).  How about just having the interface be:

```
type RepositoryFetcher {
   Fetch(...)
   Cleanup(...)
}

```

and then in the vcs.github package you can have an implementation called RepositoryFetcher.  By calling the method Fetch, we can swap out how we're actually pulling this down in the future without renaming anything. Just an idea.